### PR TITLE
Leadership Development App: Assignments View

### PR DESCRIPTION
**Pull request by cristinabaldor**
# Pull Request

## Summary & Motivation

[//]: # "When merged, this pull request will..."

Match metrics to people based on job title, region, academic year.

## Self-review

- [x] Update **due date**, **assignee**, and **priority** on our
      [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

  _If this is a same-day request, please flag that in Slack_

- [x] <kbd>Format</kbd> has been run on all modified files

- [ ] Ensure you are using the `union_dataset_join_clause()` macro for queries that employ any
      models from these datasets:
  - deanslist
  - edplan
  - iready
  - pearson
  - powerschool
  - renlearn
  - titan

## Troubleshooting

- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207363402432274

### CODE REVIEW

File: src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    {{ ref("base_people__staff_roster") }} as sr2
+    on sr.report_to_employee_number = sr2.employee_number
+where
+    ag.route = ldm.role
+    and (sr.business_unit_home_name = ldm.region or ldm.region = 'All')
+    and sr.assignment_status in ('Active', 'Leave')
+    /* Need '2024' to make visible before start of next academic year,
+    will switch after 7/1/2024*/
+    and 2024 = ldm.academic_year
+order by sr.employee_number
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:57]: remove order by

File: src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+left join
+    {{ ref("base_people__staff_roster") }} as sr2
+    on sr.report_to_employee_number = sr2.employee_number
+where
+    ag.route = ldm.role
+    and (sr.business_unit_home_name = ldm.region or ldm.region = 'All')
+    and sr.assignment_status in ('Active', 'Leave')
+    /* Need '2024' to make visible before start of next academic year,
+    will switch after 7/1/2024*/
+    and 2024 = ldm.academic_year
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:56]: move to join, reverse order

File: src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+from {{ ref("base_people__staff_roster") }} as sr
+cross join
+    {{ ref("stg_performance_management__leadership_development_metrics") }} as ldm
+left join assignment_group as ag on sr.employee_number = ag.employee_number
+left join
+    {{ ref("base_people__staff_roster") }} as sr2
+    on sr.report_to_employee_number = sr2.employee_number
+where
+    ag.route = ldm.role
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:51]: move to join

File: src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    ldm.description,
+
+    ag.route,
+
+    concat(sr.employee_number, ldm.metric_id) as assignment_id,
+
+from {{ ref("base_people__staff_roster") }} as sr
+cross join
+    {{ ref("stg_performance_management__leadership_development_metrics") }} as ldm
+left join assignment_group as ag on sr.employee_number = ag.employee_number
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:46]: make inner join

File: src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+left join
+    {{ ref("base_people__staff_roster") }} as sr2
+    on sr.report_to_employee_number = sr2.employee_number
+where
+    ag.route = ldm.role
+    and (sr.business_unit_home_name = ldm.region or ldm.region = 'All')
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:52]: move to ldm join

File: src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    ldm.bucket,
+    ldm.type,
+    ldm.description,
+
+    ag.route,
+
+    concat(sr.employee_number, ldm.metric_id) as assignment_id,
+
+from {{ ref("base_people__staff_roster") }} as sr
+cross join
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:44]: make inner

File: src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+            ) as route,
+        from {{ ref("base_people__staff_roster") }}
+        where assignment_status in ('Active', 'Leave')
+    )
+select
+
+    sr.employee_number,
+    sr.google_email,
+
+    sr2.google_email as google_email_manager,
```
Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:29]: I think this already exists as its own column on `base_people__staff_roster`

Comment (cbini) [inline src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:29]: move to CTE

**Review by cbini — CHANGES_REQUESTED** [review]
hmu if you want to talk these over together

File: src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql (hunk 7 above)
Comment (cristinabaldor) [reply src/dbt/kipptaf/models/performance_management/rpt_appsheet__leadership_development_assignments.sql:29]: is it called something else? 
image

**Pull request merged**